apps/web/src/app/_components/search/search-filter.tsx

```diff
@@ -22,8 +24,24 @@ export default function SearchFilter() {
     },
   });
 
+  const searchParams = useSearchParams();
+  const router = useRouter();
+  const pathName = usePathname();
+
+  const createQueryString = useCallback(
+    (name: string, value: string) => {
+      const params = new URLSearchParams(searchParams);
+      params.set(name, value);
+      return params.toString();
+    },
+    [searchParams],
+  );
+
   function onSubmit(values: z.infer<typeof formSchema>) {
-    console.log(values);
+    // console.log(searchParams.get("search")); query the user search info
+    router.push(
+      pathName + `/?${createQueryString("search", values.searchText)}`,
```

Comment:
We should maybe add a check here to make sure that the search string in the search bar is not empty (`""`) to make sure that we don't do an empty search. Right now, if you press *Enter* without entering anything in the search bar, the search param _is_ updated to `?search=`.